examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -0,0 +1,683 @@
+  170.0: 'packages/flutter_gallery_assets/pesto/logo_big.png',
+};
+final ThemeData _kTheme = new ThemeData(
+  brightness: ThemeBrightness.light,
+  primarySwatch: Colors.teal,
+  accentColor: Colors.redAccent[200]  // TODO(mpcomplete): Why isn't this a swatch?
+);
+const String _kFontFace = 'Raleway';
+const double _kAppBarHeight = 128.0;
+const double _kRecipePageMaxWidth = 500.0;
+
+Set<Recipe> favoriteRecipes = new Set<Recipe>();
+
+// Helper for common Pesto text style properties.
+TextStyle _textStyle(double size, [FontWeight fontWeight]) {
+  return new TextStyle(
+    inherit: false,
+    fontSize: size,
+    fontWeight: fontWeight,
+    fontFamily: 'Raleway',
+    color: Colors.black87,
+    textBaseline: TextBaseline.alphabetic
+  );
+}
+
+class PestoDemo extends StatefulWidget {
+  PestoDemo({ Key key, this.showFavorites: false }) : super(key: key);
+
+  static const String routeName = '/pesto';
+
+  final bool showFavorites;
+
+  @override
+  _PestoDemoState createState() => new _PestoDemoState();
+}
+
+class _PestoDemoState extends State<PestoDemo> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
+
+  @override
+  Widget build(BuildContext context) {
+    final double statusBarHeight = MediaQuery.of(context).padding.top;
+    final TextStyle userStyle = _textStyle(12.0, FontWeight.bold);
+    final TextStyle emailStyle = _textStyle(12.0).copyWith(color: Colors.black54);
+
+    List<Recipe> recipes = config.showFavorites ? favoriteRecipes.toList() : kRecipes;
+
+    return new Theme(
+      data: _kTheme,
+      child: new Scaffold(
+        key: _scaffoldKey,
+        appBarBehavior: AppBarBehavior.under,
+        appBar: new AppBar(
+          expandedHeight: 128.0,
+          actions: <Widget>[
+            new IconButton(
+              icon: Icons.search,
+              tooltip: 'Search',
+              onPressed: () {
+                _scaffoldKey.currentState.showSnackBar(new SnackBar(
+                  content: new Text('Not supported.')
+                ));
+              }
+            )
+          ],
+          flexibleSpace: new LayoutBuilder(
+            builder: (BuildContext context, Size size) {
+              double statusBarHeight = MediaQuery.of(context).padding.top;
+              double appBarHeight = size.height - statusBarHeight;
+              double bestHeight = _kLogoImages.keys.lastWhere(
+                (double height) => appBarHeight >= height
+              );
+              return new Padding(
+                padding: new EdgeInsets.only(top: statusBarHeight),
+                child: new Align(
+                  alignment: FractionalOffset.center,
+                  child: new AssetImage(
+                    name: _kLogoImages[bestHeight],
+                    fit: ImageFit.none
+                  )
+                )
+              );
+            }
+          )
+        ),
+        drawer: new Drawer(
+          child: new Block(children: <Widget>[
```

Comment:
I know, but I like sticking `children` on the previous line so that indentation matches the widget hierarchy. e.g.
```
  Drawer:
    Block:
      Button
      Button
```
as opposed to
```
  Drawer:
    Block:
      children:
        Button
        Button
```